core/lib/types/src/block.rs

```diff
@@ -345,4 +357,66 @@ mod tests {
         assert_eq!(block_number, unpacked_block_number);
         assert_eq!(block_timestamp, unpacked_block_timestamp);
     }
+
+    #[test]
+    fn test_build_bloom() {
+        let logs = [
+            (
+                Address::from_str("0x86Fa049857E0209aa7D9e616F7eb3b3B78ECfdb0").unwrap(),
+                vec![
+                    H256::from_str(
+                        "0x3452f51d00000000000000000000000000000000000000000000000000000000",
+                    )
+                    .unwrap(),
+                    H256::from_str(
+                        "0x000000000000000000000000d0a6e6c54dbc68db5db3a091b171a77407ff7ccf",
+                    )
+                    .unwrap(),
+                    H256::from_str(
+                        "0x0000000000000000000000000f5e378a82a55f24e88317a8fb7cd2ed8bd3873f",
+                    )
+                    .unwrap(),
+                    H256::from_str(
+                        "0x000000000000000000000000000000000000000000000004f0e6ade1e67bb719",
+                    )
+                    .unwrap(),
+                ],
+            ),
+            (
+                Address::from_str("0x86Fa049857E0209aa7D9e616F7eb3b3B78ECfdb0").unwrap(),
+                vec![
+                    H256::from_str(
+                        "0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef",
+                    )
+                    .unwrap(),
+                    H256::from_str(
+                        "0x000000000000000000000000d0a6e6c54dbc68db5db3a091b171a77407ff7ccf",
+                    )
+                    .unwrap(),
+                    H256::from_str(
+                        "0x0000000000000000000000000f5e378a82a55f24e88317a8fb7cd2ed8bd3873f",
+                    )
+                    .unwrap(),
+                ],
+            ),
+            (
+                Address::from_str("0xd0a6E6C54DbC68Db5db3A091B171A77407Ff7ccf").unwrap(),
+                vec![H256::from_str(
+                    "0x51223fdc0a25891366fb358b4af9fe3c381b1566e287c61a29d01c8a173fe4f4",
+                )
+                .unwrap()],
+            ),
+        ];
+        let iter = logs.iter().flat_map(|log| {
+            log.1
+                .iter()
+                .map(|topic| BloomInput::Raw(topic.as_bytes()))
+                .chain(iter::once(BloomInput::Raw(log.0.as_bytes())))
+        });
+
+        let bloom = build_bloom(iter);
+        let expected = Bloom::from_str("0x00000000040000000000000001000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000200000000000002000040000018000000000000000000000000000000000000000000000000000000000000000000000000000000000800000000002010000000002000000000000000400000000000080000008000000000000000000000000000000000000000000000000000000400000000000100000000000080400000000000000002000000000000000000000004000000000000000000000008002000000000000000000010000000000000000000000000000000000000000000000000").unwrap();
```

Comment:
Nit: It's possible to escape newlines / following whitespace in a string literal, so that it's more readable:

```rust
Bloom::from_str(
    "0x0000000004000000000000000100000000000000000000000000\
     000000000000000000000000000000004000000000000000000000\
     000000000000000000020000000000000200004..."
)
```